File: Data Structures & Algorithms/is-palindrome/submission-1.java
class Solution {
    public boolean isPalindrome(String s) {
        String st =s.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
       int l=0;
       int r=st.length()-1;
       while(l<r){
        if(st.charAt(l) !=st.charAt(r)){
            return false;
        }
        l++;
        r--;
       }
       return true;
        
    }
}

File: Data Structures & Algorithms/sqrtx/submission-1.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x==0:
            return x
        l=0
        r=x
        while l<r:
            m=(l+r)//2
            sq=m*m
            if sq<=x:
                l=m+1
            else:
                r=m
        return l-1            


        